src/test/java/com/google/alpollo/AnalysisTest.java

```diff
@@ -76,15 +80,38 @@ public void top10SalientEntitiesWithLessThan10Entities() throws IOException {
    */
   @Test
   public void top10SalientEntitiesWithMoreThan10Entities() throws IOException {
-    List<Entity> entityList = new ArrayList<>(AnalysisHelper.getEntityList(LYRICS_LONG));
+    List<Entity> entityList = AnalysisHelper.getEntityList(LYRICS_LONG);
     List<SongEntity> simplifiedEntityList = AnalysisHelper.getSimplifiedEntityList(entityList);
 
     String actual = gson.toJson(AnalysisHelper.getTopSalientEntities(simplifiedEntityList));
-    String expected = gson.toJson(Arrays.asList(new SongEntity("seaside", 0.36), 
-        new SongEntity("source", 0.34), new SongEntity("mountain", 0.05), new SongEntity("ones", 0.03), 
-        new SongEntity("one", 0.03), new SongEntity("roundabout", 0.02), new SongEntity("love", 0.02),
-        new SongEntity("nights", 0.02), new SongEntity("fire", 0.02), new SongEntity("root", 0.01)));
+    String expected = gson.toJson(Arrays.asList(new SongEntity("seaside", 0.36, "LOCATION", Collections.emptyMap()), 
+        new SongEntity("source", 0.34, "PERSON", Collections.emptyMap()),
+        new SongEntity("mountain", 0.05, "OTHER", Collections.emptyMap()), 
+        new SongEntity("ones", 0.03, "PERSON", Collections.emptyMap()), 
+        new SongEntity("one", 0.03, "PERSON", Collections.emptyMap()), 
+        new SongEntity("roundabout", 0.02, "OTHER", Collections.emptyMap()), 
+        new SongEntity("love", 0.02, "OTHER", Collections.emptyMap()),
+        new SongEntity("nights", 0.02, "EVENT", Collections.emptyMap()), 
+        new SongEntity("fire", 0.02, "OTHER", Collections.emptyMap()), 
+        new SongEntity("root", 0.01, "OTHER", Collections.emptyMap())));
         
     Assert.assertEquals(expected, actual);
   }
+
+  /** Some entities might have metadata attached to them. Let's see if they're shown correctly. */
+  @Test
+  public void checkEntitiesWithMetadata() throws IOException {
+    List<Entity> entityList = AnalysisHelper.getEntityList(LYRICS_WITH_METADATA);
+    List<SongEntity> simplifiedEntityList = AnalysisHelper.getSimplifiedEntityList(entityList);
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("mid", "/m/045c7b");
+    metadata.put("wikipedia_url", "https://en.wikipedia.org/wiki/Google");
+
+    String actual = gson.toJson(AnalysisHelper.getTopSalientEntities(simplifiedEntityList));
+    String expected = gson.toJson(Arrays.asList(new SongEntity("Google", 0.88, "ORGANIZATION", metadata),
+        new SongEntity("results", 0.12, "OTHER", Collections.emptyMap())));
```

Comment:
I think the best way to compare `SongEntity` objects it to override `equals` and compare objects instead of json strings, which will compare metadata as sets (unordered), so if in the future we'll add not only Wikipedia URL, but smth else, we won't need to change these tests

Reply:
I tried comparing objects and it did not really work, it always fails. Maybe I don't know how?

Reply:
how you compared them?

Reply:
`Assert.assertEquals(list, list)'
All the objects had different IDs
I'm trying now with `Assert.assertThat()`.

Reply:
Ok your idea seems the best, I could override `equals` and compare the objects in 2 for loops, to avoid converting them to strings.
Maybe the guys have a better idea? Let's check with them too.

Reply:
Also you can work not with lists of `Entities`, but with sets or another unordered collection, if the order doesn't matter, and in this case there's no need to do loops

Reply:
I think order matters in our case, we're talking about top 10 most important entities.

Reply:
so, I think that overriding `equals` and `assertThat(actual, is(expected));` should work